### ISSUE TITLE
MySQL Upgrade from 5.5 to 5.7 broke boolean columns

### ISSUE THREAD

**Issue by mameier** [brianmario/mysql2#773]
After upgrading MySQL from 5.5 to 5.7 (and reinstalling the same Version of mysql2 (0.3.17) all my boolean columns are broken. 

I use null: false, default: false for all boolean columns.

before the upgrade
`some.update_attribute :broken, false`
created a `UPDATE ... SET broken=0` statement.
Now a `UPATE ... SET broken = null` is generated which is rejected by MySQL.

**Comment by sodabrew** [comment]
Please define what `broken` means to you, I have no idea what problem you're describing.

`update_attribute` is a Rails method. Did you change versions of Rails?

The mysql2 gem does not generate SQL, Active Record does that part.

**Comment by mameier** [comment]
I found my mistake:
I assigned the boolean values true and false to attributes.
Assigning false to an attribute results in a new value of "nil" which in turn becomes a SET ... = null.
MySQL 5.5 did assign the default value if a column was not null and had a default value, while MySQL 5.7 throws an error.

The sulution is to assign string values "true" and "false" to attributes, that are typecasted to true and false which are quoted by arel to '1' and '0'.

Issue can be closed.
